### ISSUE TITLE
UCIEngine PlayerIsDead exception after a piece promotion

### ISSUE THREAD

**Issue by pychess** [pychess/pychess#547]
Original [issue 547](https://code.google.com/p/pychess/issues/detail?id=547) reported by [mattgatto](https://code.google.com/u/mattgatto/) 2010-04-05

**What steps will reproduce the problem?**
1. Load the attached pos.fen and play White against Fruit or Stockfish as Black
2. play f6f7 ... f7f8q
3. wait for engine's reply

**What is the expected output? What do you see instead?**
Excerpt, but full log is attached as well:
04:07:11 Default Error: A Player died: player=Fruit 2.1 error=Traceback (most recent call last):
  File "/home/gatto/code/svn/pychess-bugfix/lib/pychess/Utils/GameModel.py", line 396, in run
    self.boards[-2])
  File "/home/gatto/code/svn/pychess-bugfix/lib/pychess/Players/UCIEngine.py", line 184, in makeMove
    raise PlayerIsDead
PlayerIsDead

This happens because UCIEngine._startPonder doesn't convert self.pondermove to AN
before sending it in the 'position' command string to the engine:

```
def \_startPonder (self):
    print >> self.engine, "position fen", self.board.asFen(), \
                            "moves", self.pondermove
    print >> self.engine, "go ponder wtime", self.wtime, \
        "btime", self.btime, "winc", self.incr, "binc", self.incr
```

Also, when the move is sent in the 'position ... <pondermove>' string, it has
to match the format of the move string sent from the engine in the
'bestmove ... pondermove ...' string or both stockfish and fruit will either
crash or seems to misinterperet the move and then send an illegal move in reply
causing a PlayerIsDead exception to be raised. The UCI protocol page
http://wbec-ridderkerk.nl/html/UCIProtocol.html says promotions should be in the
"short" AN form:
## Move format:

The move format is in long algebraic notation.
A nullmove from the Engine to the GUI should be send as 0000.
Examples:  e2e4, e7e5, e1g1 (white short castling), e7e8q (for promotion)

The attached patch fixes the problem (but doesn't include the necessary changes to
UCIEngine._startPonder() which are included in the patch of changes attached to the
UCIEngine synchronization problems issue.

**Comment by pychess** [comment]
- attached [UCIEngine PlayerIsDead after promotion issue report.diff](https://code.google.com/p/pychess/issues/detail?id=547#c0)
- attached [pos.fen](https://code.google.com/p/pychess/issues/detail?id=547#c0)
- attached [UCIEngine PlayerIsDead after promotion issue report.log](https://code.google.com/p/pychess/issues/detail?id=547#c0)

**Comment by pychess** [comment]
Original [comment 1.](https://code.google.com/p/pychess/issues/detail?id=547#c1) posted by [lobais](https://code.google.com/u/lobais/) 2010-04-06

Yes, currently the stupid line:
position fen 8/p3BPk1/1p4P1/1P2K3/4p3/P7/8/3q4 w - - 1 59 moves f7f8
is sent.
Patch fixes this nicely.

**Comment by pychess** [comment]
Original [comment 2.](https://code.google.com/p/pychess/issues/detail?id=547#c2) posted by [mattgatto](https://code.google.com/u/mattgatto/) 2010-06-18

Fixed in revision 7c706be7b4.
- changed status to: Fixed

**Comment by pychess** [comment]
Original [comment 3.](https://code.google.com/p/pychess/issues/detail?id=547#c3) posted by [gbtami](https://code.google.com/u/gbtami/) 2010-10-23

issue #599 has been merged into this issue.
